mesonbuild/wrap/wrap.py

```diff
@@ -65,24 +69,26 @@ def quiet_git(cmd: typing.List[str], workingdir: str) -> typing.Tuple[bool, str]
 
 def open_wrapdburl(urlstring: str) -> 'http.client.HTTPResponse':
     global ssl_warning_printed
+
     if has_ssl:
+        if not urlstring.startswith(whitelist_domain):
+            raise WrapException('{} is not a whitelisted URL'.format(urlstring))
```

Comment:
There is work afoot to create wrapdb mirrors, that most likely will have names such as `mirror1.wrapdb.mesonbuild.com`. For future proofness, this code should parse the server name and validate that it is a subdomain of wrapdb.